controllers/pipelines/runconfiguration_controller_decoupled_test.go

```diff
@@ -146,4 +226,42 @@ var _ = Describe("RunConfiguration controller k8s integration", Serial, func() {
 			})).Should(Succeed())
 		})
 	})
+
+	When("More than one RunSchedule exists for a RunConfiguration", func() {
+		It("keeps only one matching RunSchedule", func() {
+			runConfiguration := pipelinesv1.RandomRunConfiguration()
+			CreateSucceeded(runConfiguration)
+
+			matchingSchedule := runScheduleForRunConfiguration(runConfiguration)
+
+			runSchedules := []*pipelinesv1.RunSchedule{
+				matchingSchedule,
+				matchingSchedule.DeepCopy(),
+				pipelinesv1.RandomRunSchedule(),
+				pipelinesv1.RandomRunSchedule(),
+			}
+
+			for _, runSchedule := range runSchedules {
+				Expect(controllerutil.SetControllerReference(runConfiguration, runSchedule, scheme.Scheme)).To(Succeed())
+				Expect(k8sClient.Create(ctx, runSchedule)).To(Succeed())
+			}
+
+			Eventually(func(g Gomega) {
+				ownedSchedules, err := findOwnedRunSchedules(ctx, k8sClient, runConfiguration)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(ownedSchedules).To(HaveLen(1))
+				g.Expect(ownedSchedules[0].Spec).To(Equal(matchingSchedule.Spec))
```

Comment:
It would be cool to have something like `g.Expect(ownedSchedules).To(HaveOneElementEqualTo(matchingSchedule.Spec))` or something like that.

Reply:
it would be nice indeed, but we would have to access a field of the array elements still.